src/vcpkg/base/system.process.cpp

```diff
@@ -676,18 +676,33 @@ namespace vcpkg
         auto rc = cmd_execute_and_stream_data(
             cmd_line,
             wd,
-            [&](StringView sv) {
-                auto prev_size = buf.size();
-                Strings::append(buf, sv);
-
-                auto it = std::find(buf.begin() + prev_size, buf.end(), '\n');
-                while (it != buf.end())
+            [&](const StringView sv) {
+                const auto last = sv.end();
+                auto start = sv.begin();
+                auto it = std::find(start, last, '\n');
+                if (it != last)
                 {
-                    std::string s(buf.begin(), it);
-                    per_line_cb(s);
-                    buf.erase(buf.begin(), it + 1);
-                    it = std::find(buf.begin(), buf.end(), '\n');
+                    // include the prefix of this line from the last callback
+                    buf.append(start, it);
+                    // omit trailing \r
+                    if (!buf.empty() && buf.back() == '\r') buf.pop_back();
+                    per_line_cb(buf);
+                    buf.clear();
+                    while (start = it + 1, it = std::find(start, last, '\n'), it != last)
+                    {
+                        if (it != start && *(it - 1) == '\r')
+                        {
+                            // omit trailing \r
+                            per_line_cb({start, it - 1});
+                        }
+                        else
+                        {
+                            per_line_cb({start, it});
+                        }
+                    }
                 }
+
+                buf.append(start, last);
```

Comment:
I think this doesn't emit the last line.

Reply:
The last line is emitted below the fold, on line 709.